quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/LoggingPropertyMappers.java

```diff
@@ -64,15 +78,96 @@ public String apply(String value, ConfigSourceInterceptorContext configSourceInt
                         .defaultValue("%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%c] (%t) %s%e%n")
-                        .description("The format of log entries. If the format has spaces in it, you need to escape the value such as \"<format>\".")
+                        .description("The format of plain console log entries. If the format has spaces in it, escape the value using \"<format>\".")
                         .paramLabel("format")
+                        .build(),
+                builder().from("log-console-color")
+                        .to("quarkus.log.console.color")
+                        .defaultValue(Boolean.TRUE.toString())
+                        .description("Enable or disable colors when logging to console.")
+                        .paramLabel(Boolean.TRUE + "|" + Boolean.FALSE)
+                        .build(),
+                builder().from("log-console-enabled")
+                        .mapFrom("log")
+                        .to("quarkus.log.console.enable")
+                        .hidden(true)
+                        .transformer(resolveLogHandler(DEFAULT_LOG_HANDLER))
+                        .build(),
+                builder().from("log-file-enabled")
+                        .mapFrom("log")
+                        .to("quarkus.log.file.enable")
+                        .hidden(true)
+                        .transformer(resolveLogHandler("file"))
+                        .build(),
+                builder().from("log-file")
+                        .to("quarkus.log.file.path")
+                        .defaultValue(DEFAULT_LOG_PATH)
+                        .description("Set the log file path and filename.")
+                        .paramLabel("<path>/<file-name>.log")
+                        .transformer(LoggingPropertyMappers::resolveFileLogLocation)
+                        .build(),
+                builder().from("log-file-format")
+                        .to("quarkus.log.file.format")
+                        .defaultValue("%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%c] (%t) %s%e%n")
+                        .description("Set a format specific to file log entries.")
+                        .paramLabel("<format>")
                         .build()
         };
     }
 
+    private static BiFunction<String, ConfigSourceInterceptorContext, String> resolveLogHandler(String handler) {
+        return (parentValue, context) -> {
+
+            //we want to fall back to console to not have nothing shown up when wrong values are set.
+            String consoleDependantErrorResult = handler.equals(DEFAULT_LOG_HANDLER) ? Boolean.TRUE.toString() : Boolean.FALSE.toString();
+
+            if(parentValue.isBlank()) {
+                addInitializationException(Messages.emptyValueForKey("log"));
+                return consoleDependantErrorResult;
+            }
+
+            String[] logHandlerValues = parentValue.split(",");
+
+            if (!AVAILABLE_LOG_HANDLERS.containsAll(List.of(logHandlerValues))) {
```

Comment:
This can happen only if someone skip the "Keycloak configuration" and try to set the underlying value using Quarkus properties? Correct?

Should we actively prevent it?

Reply:
can happen when you use a conf file or an environment variable. the `expectedValues` only prevents wrong values being set from CLI, not in general. See also the `LoggingTest`, there I check for this behaviour